esrally/driver/runner.py

```diff
@@ -1320,46 +1329,56 @@ async def _wait_for(self, es, idx, description):
 
     async def __call__(self, es, params):
         source_index = mandatory(params, "source-index", self)
+        source_indices_get = await es.indices.get(source_index)
+        source_indices = list(source_indices_get.keys())
+        source_indices_stem = commonprefix(source_indices)
+
         target_index = mandatory(params, "target-index", self)
+
         # we need to inject additional settings so we better copy the body
         target_body = deepcopy(mandatory(params, "target-body", self))
         shrink_node = params.get("shrink-node")
         # Choose a random data node if none is specified
+        node_names = [shrink_node] if shrink_node else []
```

Comment:
I think this makes the code a bit hard to read as you operate on two variables (`node_names` and `shrink_node`) in several branches and check `shrink_node` multiple times.

How about this instead?

```python
        # Choose a random data node if none is specified
        if shrink_node:
            node_names = [shrink_node]
        else:
            node_names = []
            # choose a random data node
            node_info = await es.nodes.info()
            for node in node_info["nodes"].values():
                if "data" in node["roles"]:
                    node_names.append(node["name"])
            if not node_names:
                raise exceptions.RallyAssertionError("Could not choose a suitable shrink-node automatically. Please specify it explicitly.")
```

This makes it easier to reason because there is only one top-level conditional and then we populate `node_names` in each branch separately.

Reply:
Addressed in a3cab0e5a5470d7dcdc1964d6c48c8d0a32969ab (also removed please following https://developers.google.com/style/tone#politeness-and-use-of-please)